### PR TITLE
be2 database

### LINKED ISSUE #23: Integrate database in the publish-subscribe model
Add database to the publish-subscribe model for be2.

### PR DESCRIPTION
**Pull request by SFulpius**
Integrate the database in the publish-subscribe implementation and handles notification and fetch requests.
Closes #23.

### CODE REVIEW

**Review by RaulinN — APPROVED** [review]

File: src/test/scala/pubsub/PubSubTest.scala [hunk 1]
```diff
@@ -25,12 +31,26 @@ class PubSubTest extends FunSuite {
...
     implicit val system: ActorSystem[SpawnProtocol.Command] = ActorSystem[SpawnProtocol.Command](root, "test")
     implicit val timeout = Timeout(1, TimeUnit.SECONDS)
-    val (entry, actor) = setup()
+    val (entry, actor, dbActor) = setup(DatabasePath)
     val sinkHead = Sink.head[JsonMessage]
 
-    val flows = (1 to numberProcesses).map(_ => PublishSubscribe.jsonFlow(entry, actor))
+    val options: Options = new Options()
+    options.createIfMissing(true)
+
```
Comment (lhmerino) [inline src/test/scala/pubsub/PubSubTest.scala:48]: Extra whitespace

File: src/main/scala/ch/epfl/pop/pubsub/ChannelActor.scala [hunk 2]
```diff
@@ -4,7 +4,7 @@ import akka.NotUsed
 import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.{ActorRef, Behavior}
 import akka.stream.scaladsl.{BroadcastHub, Keep, MergeHub, Sink, Source}
-import ch.epfl.pop.json.JsonMessages.{AnswerMessageServer, JsonMessage, PublishChannelClient}
+import ch.epfl.pop.json.JsonMessages.{AnswerMessageServer, JsonMessage, NotifyChannelServer, PublishChannelClient}
 
 object ChannelActor {
```
Comment (lhmerino) [inline src/main/scala/ch/epfl/pop/pubsub/ChannelActor.scala:9]: Naming inconsistent with ActorDB

**Review by lhmerino — APPROVED** [review]
Looks good! Please provide a few additional comments such as the role of ActorDB and ChannelActor (note also the difference in convention in naming).

**Comment by SFulpius** [comment]
> Looks good! Please provide a few additional comments such as the role of ActorDB and ChannelActor (note also the difference in convention in naming).

Done!

**Pull request merged**